From 7ad0014f05b68d592223f2b6c6970d06b0b3f434 Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Sun, 25 Jul 2021 13:34:55 +0300
Subject: [PATCH] [SW-45031] greco: remove duplicate headers

Instead of keeping duplicate copies of tstd.h header
in greco/arm and greco/arc, kept only a single copy in
greco/common.

Change-Id: I4d0b1070bbb10e316c03b42034601f835145ac9f
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/arch/greco/arc/tstd.h b/arch/greco/arc/tstd.h
deleted file mode 100644
index 5ba1f76..0000000
--- a/arch/greco/arc/tstd.h
+++ /dev/null
@@ -1,49 +0,0 @@
-/* SPDX-License-Identifier:     GPL-2.0+
- *
- * Copyright (C) 2020 HabanaLabs Ltd.
- */
-
-#include <greco/greco_reg_map.h>
-#include <goya2_blocks_linux_driver.h>
-#include <dcon0_tstd_masks.h>
-#include <dcon0_tstd_regs.h>
-
-/*TODO: check if the same file can be used across targets of Greco */
-#define HL_TS_CMN_SDIF_WDATA_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_WDATA_SHIFT
-#define HL_TS_CMN_SDIF_WDATA_MASK	DCON0_TSTD_TS_CMN_SDIF_SDIF_WDATA_MASK
-#define HL_TS_CMN_SDIF_ADDR_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_ADDR_SHIFT
-#define HL_TS_CMN_SDIF_ADDR_MASK	DCON0_TSTD_TS_CMN_SDIF_SDIF_ADDR_MASK
-#define HL_TS_CMN_SDIF_WRN_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_WRN_SHIFT
-#define HL_TS_CMN_SDIF_WRN_MASK		DCON0_TSTD_TS_CMN_SDIF_SDIF_WRN_MASK
-#define HL_TS_CMN_SDIF_PROG_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_PROG_SHIFT
-#define HL_TS_CMN_SDIF_PROG_MASK	DCON0_TSTD_TS_CMN_SDIF_SDIF_PROG_MASK
-#define HL_TS_CMN_SDIF_STATUS_BUSY_MASK	\
-	DCON0_TSTD_TS_CMN_SDIF_STATUS_SDIF_BUSY_MASK
-/*
- * sample_data register width is 16 bits, as per MR74137 spec resolution is
- * from 8 - 12 bits only.
- */
-#define HL_TS_00_SDIF_DATA_SMPL_DATA_MASK 0xFFF
-#define HL_TS_CMN_CLK_SYNTH_LO_SHIFT	\
-	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_LO_SHIFT
-#define HL_TS_CMN_CLK_SYNTH_LO_MASK	\
-	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_LO_MASK
-#define HL_TS_CMN_CLK_SYNTH_HI_SHIFT	\
-	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_HI_SHIFT
-#define HL_TS_CMN_CLK_SYNTH_HI_MASK	\
-	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_HI_MASK
-#define HL_TS_CMN_CLK_SYNTH_HOLD_SHIFT	\
-	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_HOLD_SHIFT
-#define HL_TS_CMN_CLK_SYNTH_HOLD_MASK	\
-	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_HOLD_MASK
-#define HL_TS_CMN_CLK_SYNTH_ENA_SHIFT	\
-	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYTH_EN_SHIFT
-#define HL_TS_CMN_CLK_SYNTH_ENA_MASK	\
-	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYTH_EN_MASK
-
-#define HL_TS_CMN_SDIF_STATUS		(mmDCON0_TSTD_TS_CMN_SDIF_STATUS & 0xFF)
-#define HL_TS_CMN_SDIF			(mmDCON0_TSTD_TS_CMN_SDIF & 0xFF)
-#define HL_TS_00_SDIF_DATA		(mmDCON0_TSTD_TS_00_SDIF_DATA & 0xFF)
-#define HL_TS_CMN_CLK_SYNTH		(mmDCON0_TSTD_TS_CMN_CLK_SYNTH & 0xFF)
-#define HL_TS_00_ALARMA_CFG		(mmDCON0_TSTD_TS_00_ALARMA_CFG & 0xFF)
-#define HL_TS_00_ALARMB_CFG		(mmDCON0_TSTD_TS_00_ALARMB_CFG & 0xFF)
diff --git a/arch/greco/arm/tstd.h b/arch/greco/arm/tstd.h
deleted file mode 100644
index 5ba1f76..0000000
--- a/arch/greco/arm/tstd.h
+++ /dev/null
@@ -1,49 +0,0 @@
-/* SPDX-License-Identifier:     GPL-2.0+
- *
- * Copyright (C) 2020 HabanaLabs Ltd.
- */
-
-#include <greco/greco_reg_map.h>
-#include <goya2_blocks_linux_driver.h>
-#include <dcon0_tstd_masks.h>
-#include <dcon0_tstd_regs.h>
-
-/*TODO: check if the same file can be used across targets of Greco */
-#define HL_TS_CMN_SDIF_WDATA_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_WDATA_SHIFT
-#define HL_TS_CMN_SDIF_WDATA_MASK	DCON0_TSTD_TS_CMN_SDIF_SDIF_WDATA_MASK
-#define HL_TS_CMN_SDIF_ADDR_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_ADDR_SHIFT
-#define HL_TS_CMN_SDIF_ADDR_MASK	DCON0_TSTD_TS_CMN_SDIF_SDIF_ADDR_MASK
-#define HL_TS_CMN_SDIF_WRN_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_WRN_SHIFT
-#define HL_TS_CMN_SDIF_WRN_MASK		DCON0_TSTD_TS_CMN_SDIF_SDIF_WRN_MASK
-#define HL_TS_CMN_SDIF_PROG_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_PROG_SHIFT
-#define HL_TS_CMN_SDIF_PROG_MASK	DCON0_TSTD_TS_CMN_SDIF_SDIF_PROG_MASK
-#define HL_TS_CMN_SDIF_STATUS_BUSY_MASK	\
-	DCON0_TSTD_TS_CMN_SDIF_STATUS_SDIF_BUSY_MASK
-/*
- * sample_data register width is 16 bits, as per MR74137 spec resolution is
- * from 8 - 12 bits only.
- */
-#define HL_TS_00_SDIF_DATA_SMPL_DATA_MASK 0xFFF
-#define HL_TS_CMN_CLK_SYNTH_LO_SHIFT	\
-	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_LO_SHIFT
-#define HL_TS_CMN_CLK_SYNTH_LO_MASK	\
-	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_LO_MASK
-#define HL_TS_CMN_CLK_SYNTH_HI_SHIFT	\
-	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_HI_SHIFT
-#define HL_TS_CMN_CLK_SYNTH_HI_MASK	\
-	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_HI_MASK
-#define HL_TS_CMN_CLK_SYNTH_HOLD_SHIFT	\
-	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_HOLD_SHIFT
-#define HL_TS_CMN_CLK_SYNTH_HOLD_MASK	\
-	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_HOLD_MASK
-#define HL_TS_CMN_CLK_SYNTH_ENA_SHIFT	\
-	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYTH_EN_SHIFT
-#define HL_TS_CMN_CLK_SYNTH_ENA_MASK	\
-	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYTH_EN_MASK
-
-#define HL_TS_CMN_SDIF_STATUS		(mmDCON0_TSTD_TS_CMN_SDIF_STATUS & 0xFF)
-#define HL_TS_CMN_SDIF			(mmDCON0_TSTD_TS_CMN_SDIF & 0xFF)
-#define HL_TS_00_SDIF_DATA		(mmDCON0_TSTD_TS_00_SDIF_DATA & 0xFF)
-#define HL_TS_CMN_CLK_SYNTH		(mmDCON0_TSTD_TS_CMN_CLK_SYNTH & 0xFF)
-#define HL_TS_00_ALARMA_CFG		(mmDCON0_TSTD_TS_00_ALARMA_CFG & 0xFF)
-#define HL_TS_00_ALARMB_CFG		(mmDCON0_TSTD_TS_00_ALARMB_CFG & 0xFF)
diff --git a/arch/greco/common/tstd.h b/arch/greco/common/tstd.h
new file mode 100644
index 0000000..5ba1f76
--- /dev/null
+++ b/arch/greco/common/tstd.h
@@ -0,0 +1,49 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2020 HabanaLabs Ltd.
+ */
+
+#include <greco/greco_reg_map.h>
+#include <goya2_blocks_linux_driver.h>
+#include <dcon0_tstd_masks.h>
+#include <dcon0_tstd_regs.h>
+
+/*TODO: check if the same file can be used across targets of Greco */
+#define HL_TS_CMN_SDIF_WDATA_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_WDATA_SHIFT
+#define HL_TS_CMN_SDIF_WDATA_MASK	DCON0_TSTD_TS_CMN_SDIF_SDIF_WDATA_MASK
+#define HL_TS_CMN_SDIF_ADDR_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_ADDR_SHIFT
+#define HL_TS_CMN_SDIF_ADDR_MASK	DCON0_TSTD_TS_CMN_SDIF_SDIF_ADDR_MASK
+#define HL_TS_CMN_SDIF_WRN_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_WRN_SHIFT
+#define HL_TS_CMN_SDIF_WRN_MASK		DCON0_TSTD_TS_CMN_SDIF_SDIF_WRN_MASK
+#define HL_TS_CMN_SDIF_PROG_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_PROG_SHIFT
+#define HL_TS_CMN_SDIF_PROG_MASK	DCON0_TSTD_TS_CMN_SDIF_SDIF_PROG_MASK
+#define HL_TS_CMN_SDIF_STATUS_BUSY_MASK	\
+	DCON0_TSTD_TS_CMN_SDIF_STATUS_SDIF_BUSY_MASK
+/*
+ * sample_data register width is 16 bits, as per MR74137 spec resolution is
+ * from 8 - 12 bits only.
+ */
+#define HL_TS_00_SDIF_DATA_SMPL_DATA_MASK 0xFFF
+#define HL_TS_CMN_CLK_SYNTH_LO_SHIFT	\
+	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_LO_SHIFT
+#define HL_TS_CMN_CLK_SYNTH_LO_MASK	\
+	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_LO_MASK
+#define HL_TS_CMN_CLK_SYNTH_HI_SHIFT	\
+	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_HI_SHIFT
+#define HL_TS_CMN_CLK_SYNTH_HI_MASK	\
+	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_HI_MASK
+#define HL_TS_CMN_CLK_SYNTH_HOLD_SHIFT	\
+	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_HOLD_SHIFT
+#define HL_TS_CMN_CLK_SYNTH_HOLD_MASK	\
+	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_HOLD_MASK
+#define HL_TS_CMN_CLK_SYNTH_ENA_SHIFT	\
+	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYTH_EN_SHIFT
+#define HL_TS_CMN_CLK_SYNTH_ENA_MASK	\
+	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYTH_EN_MASK
+
+#define HL_TS_CMN_SDIF_STATUS		(mmDCON0_TSTD_TS_CMN_SDIF_STATUS & 0xFF)
+#define HL_TS_CMN_SDIF			(mmDCON0_TSTD_TS_CMN_SDIF & 0xFF)
+#define HL_TS_00_SDIF_DATA		(mmDCON0_TSTD_TS_00_SDIF_DATA & 0xFF)
+#define HL_TS_CMN_CLK_SYNTH		(mmDCON0_TSTD_TS_CMN_CLK_SYNTH & 0xFF)
+#define HL_TS_00_ALARMA_CFG		(mmDCON0_TSTD_TS_00_ALARMA_CFG & 0xFF)
+#define HL_TS_00_ALARMB_CFG		(mmDCON0_TSTD_TS_00_ALARMB_CFG & 0xFF)
